blobber/amazons3_backend.py

```diff
@@ -55,6 +55,6 @@ def upload_to_AmazonS3(hashalgo, blobhash, data_file, headers, metadata):
             key.set_acl('public-read')
 
     # return the blob URL
-    amazon_baseurl = "http://" + BUCKET + ".s3.amazonaws.com"
+    amazon_baseurl = key.generate_url(expires_in=0)
```

Comment:
I think this should just be `return key.generate_url(expires_in=0)` (removing the code below)